src/codec/builtin/BigDecimalCodec.ts

```diff
@@ -3,44 +3,52 @@ import {CodecUtil} from './CodecUtil';
 import {FixSizedTypesCodec} from './FixSizedTypesCodec';
 import {BitsUtil} from '../../util/BitsUtil';
 
-function bytes2hexString(uint8array: Uint8Array): string { // buffer is an ArrayBuffer
-    return '0x' + [...uint8array].map(x => x.toString(16).padStart(2, '0')).join('');
-}
-
 /** @internal */
 export class BigDecimalCodec {
 
     static decode(clientMessage: ClientMessage): string {
         const buffer = clientMessage.nextFrame().content;
         const contentSize = FixSizedTypesCodec.decodeInt(buffer, 0);
-        const body = new Uint8Array(buffer.slice(BitsUtil.INT_SIZE_IN_BYTES, BitsUtil.INT_SIZE_IN_BYTES + contentSize));
-        const scale = FixSizedTypesCodec.decodeInt(buffer, BitsUtil.INT_SIZE_IN_BYTES + contentSize);
+        let body = buffer.slice(BitsUtil.INT_SIZE_IN_BYTES, BitsUtil.INT_SIZE_IN_BYTES + contentSize);
+        const byteArray = new Uint8Array(body);
+        const signBit = body.slice(0, 1).readUInt8(0);
```

Comment:
Can we do it without slicing the body?

Reply:
yes. I was trying to get first bit but only bytes can be sliced